File: Programmers/LV1/정리 필요/최대공약수와 최소공배수.py
# from fractions import gcd
#
# def solution(n,m):
#     answer = [gcd(n,m), n*m / gcd(n,m)]
#     return answer
#
#
# print(solution(2,5))

#유클리드 호제법

def gcd(a,b):
    c,d = max(a,b) , min(a,b)
    t = 1
    while t>0:
        t = c % d
        c,d = d , t

    answer = [c, int(a*b/c)]


    return answer

print(gcd(15,12))

File: Programmers/LV1/JAVA/최대공약수와최소공배수.java
package Programmers.level1;

public class 최대공약수와최소공배수 {

	public static void main(String[] args) {

		int n = 3;
		int m = 12;
		solution(n, m);
	}

	public static int[] solution(int n, int m) {
		int[] answer = {1,2};

		int big = Math.max(n, m);
		int small = Math.min(m, n);

		answer[0] = gcd(big, small);
		answer[1] = big * small / answer[0];

		return answer;
	}

	public static int gcd(int big, int small) {
		if (big % small == 0) {
			return small;
		} else {
			return gcd(small, big % small);
		}
	}

}
